plugins/outputs/tcp_forwarder/tcp_forwarder.go

```diff
@@ -0,0 +1,159 @@
+package tcp_forwarder
+
+import (
+	"errors"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/serializers"
+	"io"
+	"log"
+	"net"
+	"strings"
+	"sync"
+	"time"
+)
+
+// TCPForwarder structure for configuration and server
+type TCPForwarder struct {
+	sync.Mutex
+
+	Server     string
+	Timeout    string
```

Comment:
change the type of `Timeout` to `internal.Duration`, then you don't need to parse it, the toml parsing will deal with verifying that it's a valid timestamp.

in the config file it will still look like `timeout = "5s"`